src/ipa_pc/data_structures.rs

```diff
@@ -129,8 +133,12 @@ impl<G: AffineCurve> PCVerifierKey for VerifierKey<G> {
     /// Randomize key for testing purpose
     fn randomize(&mut self) {
         let mut rng = thread_rng();
-        self.comm_key = self.comm_key.iter().map(|_| {
-            G::Projective::rand(&mut rng).into_affine()
+        self.comm_key = self.comm_key.iter().enumerate().map(|(index, &v)| {
+            if index == 0 {
+                v
+            } else {
+                G::Projective::rand(&mut rng).into_affine()
+            }
```

Comment:
Why is the first element left unchanged?

Reply:
Actually now this not needed any more, I have reverted this change